azurerm/internal/services/datafactory/data_factory_linked_service_sql_server_resource.go

```diff
@@ -55,11 +54,36 @@ func resourceDataFactoryLinkedServiceSQLServer() *pluginsdk.Resource {
 
 			"connection_string": {
 				Type:             pluginsdk.TypeString,
-				Required:         true,
+				Optional:         true,
+				AtLeastOneOf:     []string{"connection_string", "key_vault_connection_string"},
+				ConflictsWith:    []string{"key_vault_connection_string"},
```

Comment:
Do you mean `ExactlyOneOf`?
```suggestion
				ExactlyOneOf:     []string{"connection_string", "key_vault_connection_string"},
```

Reply:
Oh yes, thank you. Got lost in all those :) will push in a sec.